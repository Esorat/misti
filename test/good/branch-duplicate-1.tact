fun test() {
    let a: Int = 42;
    if (a == 42) {
        a = 43; // bad
    } else {
        a = 43;
    }
}
