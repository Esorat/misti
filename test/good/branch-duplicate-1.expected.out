[HIGH] BranchDuplicate: Duplicated code in conditional branches is detected
test/good/branch-duplicate-1.tact:3:5:
  2 |     let a: Int = 42;
> 3 |     if (a == 42) {
          ^
  4 |         a = 43; // bad
Help: Identical code in both branches detected. Refactor to eliminate redundancy.
See: https://nowarp.io/tools/misti/docs/detectors/BranchDuplicate